KLSadd_insertion/src/updateChapters.py

```diff
@@ -114,48 +371,74 @@ def findReferences(chapter):
-            for unit in mathPeople:
-                subunit = unit[unit.find(" ") + 1: unit.find("#")]
+            w = word[word.find("{")+1: word.find("}")]
+            ws = word[word.find("{")+1: word.find("~")]
+            for unit in math_people:
+                subunit = unit[unit.find(" ")+1: unit.find("#")]
                 # System of checks that verifies if section is in chapter
-                if ((w in subunit) or (ws in subunit)) and (chaptercheck in unit) and (len(w) == len(
-                        subunit)) or (("Pseudo Jacobi" in w) and ("Pseudo Jacobi (or Routh-Romanovski)" in subunit)):
-                    canAdd = True
+                if (w in subunit or ws in subunit) and (chaptercheck in unit) and len(w) == len(subunit) or\
+                ("Pseudo Jacobi" in w and "Pseudo Jacobi (or Routh-Romanovski)" in subunit):
+                    canadd = True
                     if chapticker == 0:
-                        ref9II.append(index)
-                        ref9III.append(index)
+                        ref9_2.append(index)
+                        ref9_3.append(index)
                     elif chapticker == 1:
-                        ref14II.append(index)
-                        ref14III.append(index)
-        if("\\subsection*{References}" in word) and (canAdd):
+                        ref14_2.append(index)
+                        ref14_3.append(index)
+                    specialdetector = 0
+        if"\\subsection*{References}" in word and canadd:
             # Appends valid locations
             references.append(index)
             if chapticker == 0:
-                ref9II.append(index)
-                ref9III.append(index)
+                ref9_2.append(index)
+                ref9_3.append(index)
             elif chapticker == 1:
-                ref14II.append(index)
-                ref14III.append(index)
-            canAdd = False
-        if ("subsection*{" in word and "References" not in word):
+                ref14_2.append(index)
+                ref14_3.append(index)
+            canadd = False
+        if "subsection*{" in word and "References" not in word:
             if chapticker == 0:
-                ref9III.append(index)
+                ref9_3.append(index)
             elif chapticker == 1:
-                ref14III.append(index)
-    print(ref9II)
-    print (ref14II)
-    print(ref9III)
-    print(ref14III)
+                ref14_3.append(index)
+        if "\\section{" in word and specialdetector == 1 and chaptercheck == "14":
+            w2 = word[word.find("{") + 1: word.find("}")]
+            if "Bessel" not in w2:
+                ref14_3.append(index)
+    besselcheck = 1
+    bqlegendrecheck = 2
+
+    for i in ref9_3:
+        if i == 2646:
+            besselcheck = 0
+    if besselcheck == 1:
+        ref9_3.insert(230, 2646)
+    for i in ref14_3:
+        if i == 1217:
+            bqlegendrecheck -= 1
+    if bqlegendrecheck > 0:
+        pass
```

Comment:
I think this should be a `continue`, but even then it doesn't really make any sense, seeing as there are no conditions following it to be skipped.